src/utils.js

```diff
@@ -33,20 +33,16 @@ export function partition(array, n) {
  * @param  {Function} fn   The mapping function.
  * @return {Promise}
  */
-export function pMap(list, fn) {
+export async function pMap(list, fn) {
   let results = [];
-  return list
-    .reduce(
-      (promise, entry) => {
-        return promise.then(() => {
-          return Promise.resolve(fn(entry)).then(
-            result => results = results.concat(result)
-          );
-        });
-      },
-      Promise.resolve()
-    )
-    .then(() => results);
+  await list.reduce(
+    async function(promise, entry) {
+      await promise;
+      results = results.concat(await fn(entry));
+    },
+    Promise.resolve()
+  );
+  return results;
```

Comment:
This might be outside the scope of this PR, but this function bothers me. It seems like it has a generic, mathy-sounding name, but we only actually use it in one very specific place (`_batchRequests`). Despite the simple name `pMap`, it's actually (to use Haskell terminology) a combined (map . bind) + sequence + concat. Despite the clean name `pMap`, there's a lot of implicit ordering and implementation details present. Maybe future work can move this function from `utils.pMap` to `base.makeChunkRequests` or something.